portfolio/src/main/java/com/google/sps/libraries/AuthServlet.java

```diff
@@ -0,0 +1,57 @@
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.libraries.authstatus.AuthStatus;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/auth")
+public class AuthServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    response.setContentType("application/json;");
+
+    AuthStatus authStatus = new AuthStatus();
+
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      String urlToRedirectToAfterUserLogsOut = "/";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+
+      // response.getWriter().println("<p>Hello " + userEmail + "!</p>");
+      // response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+
+      authStatus.setUrl(logoutUrl);
+      authStatus.setVerificationStatus(true);
+
+      response.getWriter().println(new Gson().toJson(authStatus));
+    } else {
+      String urlToRedirectToAfterUserLogsIn = "/";
+      String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+
+      // response.getWriter().println("<p>Hello stranger.</p>");
+      // response.getWriter().println("<p>Login <a href=\"" + loginUrl + "\">here</a>.</p>");
+      authStatus.setUrl(loginUrl);
+      authStatus.setVerificationStatus(false);
```

Comment:
What does setVerificationStatus do? Why false on login, but true on logout, seems backwards?

Reply:
setVerificationStatus() is a function from AuthStatus that sets it's bool attribute. So in the if there's the statement `userService.isUserLoggedIn()`, if it's true then the verificationStatus is set to true and a new Logout link is given to the front end and set in the portfolio.